core/src/text/textBuffer.cpp

```diff
@@ -33,18 +33,45 @@ bool TextBuffer::addLabel(const std::string& _text, Label::Transform _transform,
         return false;
     }
 
-    std::string text = _text;
+    const std::string* text = &_text;
+    std::string transformedText;
 
-    // captilize the string
-    if (_params.uppercase) {
+    if (_params.transform != TextTransform::none) {
+        transformedText = _text;
         std::locale loc;
-        for (std::string::size_type i = 0; i < _text.length(); ++i) {
-            text[i] = std::toupper(_text[i], loc);
+
+        // perfom text transforms
+        switch (_params.transform) {
+            case TextTransform::capitalize:
+                transformedText[0] = toupper(_text[0], loc);
```

Comment:
It would toupper on a whitespace, which I guess does nothing. Noticed it's missing `std::`.